catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -442,58 +510,54 @@ private void assertReplies(List<Mutable> replies) {
     }
 
     private CatalogFramework createFramework() {
-        final long COUNT = 2;
+        return createFramework(2L, 2, 0);
+    }
+
+    private CatalogFramework createFramework(long totalHits, int... pagingResults) {
 
         CatalogFramework framework = mock(CatalogFramework.class);
-        List<Result> results = new ArrayList<Result>();
-
-        for (int i = 0; i < COUNT; i++) {
-            Result result = mock(Result.class);
-
-            MetacardImpl metacard = new MetacardImpl();
-            metacard.setId("Metacard_" + i);
-            metacard.setTitle("Metacard " + i);
-            metacard.setLocation("POINT(" + i + " " + i + ")");
-            metacard.setType(BasicTypes.BASIC_METACARD);
-            metacard.setCreatedDate(TIMESTAMP);
-            metacard.setEffectiveDate(TIMESTAMP);
-            metacard.setExpirationDate(TIMESTAMP);
-            metacard.setModifiedDate(TIMESTAMP);
-            metacard.setContentTypeName("TEST");
-            metacard.setContentTypeVersion("1.0");
-            metacard.setTargetNamespace(URI.create(getClass().getPackage()
-                    .getName()));
-
-            when(result.getDistanceInMeters()).thenReturn(100.0 * i);
-            when(result.getRelevanceScore()).thenReturn(100.0 * (COUNT - i) / COUNT);
-            when(result.getMetacard()).thenReturn(metacard);
-
-            results.add(result);
-        }
 
-        QueryResponse response = new QueryResponseImpl(mock(QueryRequest.class),
-                new ArrayList<Result>(),
-                COUNT);
-        response.getResults()
-                .addAll(results);
+        List<QueryResponse> responseList = new ArrayList<>();
+        for (int resultsPerPage : pagingResults) {
+            List<Result> results = generateResults(resultsPerPage);
+            QueryResponse response = new QueryResponseImpl(mock(QueryRequest.class),
+                    new ArrayList<>(),
+                    totalHits);
+            response.getResults()
```

Comment:
❓ Could we initialize the `results` first and pass that in to the `QueryResponseImpl` constructor instead of giving an empty `ArrayList` and then populating it? Might make the code a little more straightforward.